### ISSUE TITLE
The viewBox attribute of svg file is lost 

### ISSUE THREAD

**Issue by jairzh** [posthtml/posthtml#146]
The `viewBox` attribute of svg file is lost after using the posthtml in my webpack config, how can I prevent this behavior?

**Comment by voischev** [comment]
Show example plz.
@kisenka fix it?

**Comment by jairzh** [comment]
I used the following code to remove **Fill** attribute of svg, it works but that will also remove the **viewBox** attribute.

```javascript
var removeAttrs = require('posthtml-remove-attributes');

module.exports = {
  module: {
    loaders: [
      {
        test: /\.svg$/,
        loader: 'svg-sprite!posthtml'
      }
    ]
  },

  posthtml: function() {
    return {
      defaults: [
        removeAttrs(['fill'])
      ]
    }
  }
};
```

My svg file:

```xml
<?xml version="1.0"?>
<svg xmlns="http://www.w3.org/2000/svg" width="52" height="52" viewBox="0 0 52 52">
     <path fill="#fff" d="m46 9h-40c-2.7 0-5 2.2-5 5v24c0 2.7 2.3 5 5 5h40c2.8 0 5-2.2 5-5v-24c0-2.8-2.2-5-5-5z m-21.2 28.2h-14.4c-1.6 0-2.8-1.7-2.8-3.4 0.1-2.5 2.7-4 5.4-5.2 1.9-0.8 2.2-1.6 2.2-2.4 0-0.8-0.5-1.6-1.2-2.2-1.1-1-1.7-2.5-1.7-4.1 0-3.2 1.9-5.8 5.2-5.8s5.2 2.7 5.2 5.8c0 1.7-0.6 3.2-1.7 4.1-0.7 0.6-1.2 1.3-1.2 2.2 0 0.8 0.2 1.6 2.2 2.3 2.7 1.2 5.3 2.8 5.4 5.3 0.3 1.7-1 3.4-2.6 3.4z m19.5-5.8c0 0.9-0.8 1.7-1.7 1.7h-7.5c-0.9 0-1.7-0.7-1.7-1.7v-2.5c0-0.9 0.8-1.7 1.7-1.7h7.5c0.9 0 1.7 0.7 1.7 1.7v2.5z m0-9.1c0 0.9-0.8 1.7-1.7 1.7h-12.4c-0.9 0-1.7-0.7-1.7-1.7v-2.5c0-0.9 0.8-1.7 1.7-1.7h12.5c0.9 0 1.7 0.7 1.7 1.7v2.5z">
     </path>
</svg>
```

The result on the page:

```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" style="position:absolute;width:0;height:0;visibility:hidden">
  <defs>
    <symbol id="add_contact">
      <path fill="#fff" d="m46 9h-40c-2.7 0-5 2.2-5 5v24c0 2.7 2.3 5 5 5h40c2.8 0 5-2.2 5-5v-24c0-2.8-2.2-5-5-5z m-21.2 28.2h-14.4c-1.6 0-2.8-1.7-2.8-3.4 0.1-2.5 2.7-4 5.4-5.2 1.9-0.8 2.2-1.6 2.2-2.4 0-0.8-0.5-1.6-1.2-2.2-1.1-1-1.7-2.5-1.7-4.1 0-3.2 1.9-5.8 5.2-5.8s5.2 2.7 5.2 5.8c0 1.7-0.6 3.2-1.7 4.1-0.7 0.6-1.2 1.3-1.2 2.2 0 0.8 0.2 1.6 2.2 2.3 2.7 1.2 5.3 2.8 5.4 5.3 0.3 1.7-1 3.4-2.6 3.4z m19.5-5.8c0 0.9-0.8 1.7-1.7 1.7h-7.5c-0.9 0-1.7-0.7-1.7-1.7v-2.5c0-0.9 0.8-1.7 1.7-1.7h7.5c0.9 0 1.7 0.7 1.7 1.7v2.5z m0-9.1c0 0.9-0.8 1.7-1.7 1.7h-12.4c-0.9 0-1.7-0.7-1.7-1.7v-2.5c0-0.9 0.8-1.7 1.7-1.7h12.5c0.9 0 1.7 0.7 1.7 1.7v2.5z">
    </symbol>
  </defs>
</svg>
```

**Comment by kisenka** [comment]
@voischev [workaround it](https://github.com/kisenka/svg-sprite-loader/commit/097fa74fea52342792f3cacc7c9533aa82390fa9) :)

Test case vith mocha:

```js
var assert = require('assert');
var posthtml = require('posthtml');

it('should do not lowercase SVG attributes', function(done) {
  var source = '<svg viewBox="0 0 100 100"></svg>';
  var expected = source;

  posthtml().process(source)
    .then(function (result) {
      assert.equal(expected, result.html);
      done();
    })
    .catch(function (err) {
      done(err);
    });
});
```

I think It's a bug/feature of htmlparser2.

**Comment by voischev** [comment]
@kisenka yes. htmlparser2 returns to lowercase attribute. but it is not the reason a lost this from the result :( 

@jairzh need more info about your env. 

a simple example. its works
```js
var posthtml = require('posthtml'),
    html,
    result;

html = '<svg viewBox="10 10 0 0"></svg>';
result = posthtml().process(html, { sync: true });

console.log(result.html); // <svg viewbox="10 10 0 0"></svg>
```

**Comment by voischev** [comment]
close this issue?

**Comment by kisenka** [comment]
@voischev 
> but it is not the reason a lost this from the result :(

Because sprite-loader copy only some attributes from source SVG ([viewBox, id, class, preserveAspectRatio](https://github.com/kisenka/svg-sprite-loader/blob/master/lib/svg-document.js#L46))  and skips any other, including lowercased. I made a [workaround](https://github.com/kisenka/svg-sprite-loader/blob/master/lib/svg-document.js#L48) for copying lowercased attributes, but it's a hack.

Do you plan to support similar workaround, because now it's impossible to process camel cased tag attributes via posthtml.

**Comment by voischev** [comment]
for that I want to write own parser...

**Comment by kisenka** [comment]
@voischev any ETA? :)

**Comment by cwardzala** [comment]
I am still having issues with `viewBox` being removed from my inline SVG in my HTML file. I have passed my own options via `.posthtmlrc` but I am still seeing the attribute removed on production build? I have looked into it and it appears `HTMLAsset.js` is calling `posthtml-parser` with its own config of `{lowerCaseAttributeNames: true}` which causes `viewBox` to be removed. Why doesn't `HTMLAsset.js` use the config I am providing?
